src/tike/communicators/pool.py

```diff
@@ -94,27 +105,94 @@ def f(worker):
 
         return list(self.map(f, self.workers))
 
-    def scatter(self, x):
-        """Split x along 0th dimension and send chunks to workers`."""
-
-        def f(worker, chunk):
-            return self._copy_to(chunk, worker)
+    def scatter(self, x, s=1) -> list:
+        """Split x along 0th dimension and send chunks to workers`.
 
-        return self.map(f, self.workers, x)
+        Parameters
+        ----------
+        x : list
+            Chunks to be sent to other devices.
+        s : int
+            The size of a device group.
+            e.g., s=4 and num_gpu=8, then x[0] will be scattered
+            to workers[:4] while x[1] will go to workers[4:].
 
-    def reduce_gpu(self, x: list, worker=None) -> cp.array:
-        """Reduce x by addition to one GPU from all other GPUs."""
-        if self.num_workers == 1:
-            return x[0]
-        worker = self.workers[0] if worker is None else worker
-        with cp.cuda.Device(worker):
+        """
+        def f(worker):
+            idx = self.workers.index(worker) // s
+            return self._copy_to(x[idx], worker)
+
+        return self.map(f, self.workers)
+
+    def scatter_bcast(self, x: list, stride=1):
```

Comment:
Yes, use extended bcast() and scatter() can substitute scatter_bcast(). But I thought we agreed that we keep this separate function in case we want to decouple the ptycho and lamino APIs. I can delete it if you're not OK for keeping it in the codebase.